sw/device/exts/common/flash_crt.S

```diff
@@ -117,13 +117,5 @@ init_array_loop_end:
   li   a1, 0x0  // argv = NULL
   call main
 
-  // Load `kDeviceStopAddress` into `t0`. If it is non-zero, write the return
-  // value from `main` (in `a0`) into it, otherwise, skip to `wfi` loop.
-  lw t0, kDeviceStopAddress
-  beqz t0, wfi_loop
-  sw a0, 0(t0)
-
-  // Loop forever if main somehow returns.
-wfi_loop:
-  wfi
-  j wfi_loop
+  // Halt the core (either a wfi loop, or writing to a simulation address)
```

Comment:
Also do the same here somehow?

Reply:
Again, i don't want a call to testing-only code here, and all the flash images should call test_status_set before reaching the end of `main` anyway.